conans/client/command.py

```diff
@@ -312,6 +312,9 @@ def install(self, *args):
             try:
                 reference = ConanFileReference.loads(args.path_or_reference)
             except ConanException:
+                reference = None
```

Comment:
Otherwise, handling directly in the except, when an error occurs in the install, the trace is pretty ugly, as a "something failed when managing failure..."